lib/elixir/src/elixir_tokenizer.erl

```diff
@@ -790,7 +794,11 @@ tokenize_number([H | T], Acc, Length, Bool) when ?is_digit(H) ->
 
 %% Cast to float...
 tokenize_number(Rest, Acc, Length, true) ->
-  {Rest, list_to_float(lists:reverse(Acc)), Length};
+  try
+    {Rest, list_to_float(lists:reverse(Acc)), Length}
+  catch
+    error:badarg -> {error, "float number out of range ", lists:reverse(Acc)}
```

Comment:
If you like, otherwise we can merge this and fix it separately. The boolean blindness in the tokenize_number is not so nice.